### ISSUE TITLE
TS 5.6 `files` missing from `--showConfig` (with absolute / ``${configDir}`` `include` paths)

### ISSUE THREAD

**Issue by stabpd** [microsoft/TypeScript#60199]
### 🔎 Search Terms

"--showConfig", "include", "configDir", "files", "extends", "absolute"

### 🕗 Version & Regression Information

- Tested versions:
  - `3.2.1` (`--showConfig` was introduced here)
  - `3.2.4`
  - `5.5.2` (`${configDir}` was introduced here)
  - `latest` => `5.6.3`
  - `next` =>`5.7.0-dev.20241011`
- This is the behaviour in every version I tried, and I reviewed the FAQ for entries about "configDir", "showConfig", "absolute".
- I was unable to test this on prior versions because `--showConfig` cli flag was introduced in TS 3.2 and `configDir` template variable was introduced in TS 5.5.

### 💻 Code

```jsonc
// tsconfig-rel.json
{
    //...
    "include": ["src/**/*"]
}
```
```jsonc
// tsconfig-abs.json
{
    //...
    "include": ["/path/to/src/**/*"]
}
```
```jsonc
// tsconfig-cnf.json
{
    //...
    "include": ["${configDir}/src/**/*"]
}
```
```ts
// src/index.ts
console.log("dummy");
```

### 🙁 Actual behavior

❌  **Printing config** yields different results.

```sh
npx tsc --project tsconfig-rel.json --showConfig
npx tsc --project tsconfig-abs.json --showConfig
npx tsc --project tsconfig-cnf.json --showConfig
```

- For absolute include paths the **`files` property is missing**.
- `${configDir}` is resolved to and behaves / fails the same as absolute paths.

```diff
  {
      "compilerOptions": {...},
-     "files": [
-         "./src/index.ts"
-     ],
      "include": [
-         "src/**/*"
+         "/path/to/src/**/*"
      ],
      "exclude": [...]
  }
```

✅ **Listing files** yields the same output.
```sh
npx tsc --project tsconfig-rel.json --listFiles --noEmit
npx tsc --project tsconfig-abs.json --listFiles --noEmit
npx tsc --project tsconfig-cnf.json --listFiles --noEmit
```

List of files printed to stdout, containing
- `/path/to/`**`src/index.ts`**
- `/path/to/`**`node_modules/**/*.d.ts`** entries from npm dependency type declarations.
- `/path/to/`**`node_modules/typescript/lib/lib.decorators.d.ts`** (only in TS 5 of course)
- `/path/to/`**`node_modules/typescript/lib/lib.decorators.legacy.d.ts`** (only in TS 5 of course)

✅ **Building** yields the same `index.js` file content for all configs.
```sh
npx tsc --project tsconfig-rel.json
npx tsc --project tsconfig-abs.json
npx tsc --project tsconfig-cnf.json
```

### 🙂 Expected behavior

`--showConfig` should show the same `files` for configs that build the same files and show the same listing for `--listFiles`.

### Additional information about the issue

`includeRe` in [`matchesSpecs`](https://github.com/microsoft/TypeScript/blob/a53c37d59aa0c20f566dec7e5498f05afe45dc6b/src/compiler/commandLineParser.ts#L2656) seems to be calculated differently for absolute / relative paths, resulting in all files being filtered out in [`convertToTSConfig`](https://github.com/microsoft/TypeScript/blob/a53c37d59aa0c20f566dec7e5498f05afe45dc6b/src/compiler/commandLineParser.ts#L2588C17-L2588C34).

I can't tell though which behaviour is actually expected. (keep files or filter files out?)

**Comment by RyanCavanaugh** [comment]
I don't know if anyone is really putting absolute paths in their config files; this seems like a nonstandard usage. `showConfig` is intended for human consumption so it's not really wrong for it to do different things given different inputs. Not sure what the "best" behavior is but if someone really needs to change it to some provably better outcome, they can.

**Comment by stabpd** [comment]
I agree that this is not an urgent matter. Let me add a little clarification for future discussion though:

For me, it's not about manually configured absolute paths, but about `${configDir}` which has the same issue, because the template variable gets substituted with absolute paths internally.

When switching from a monolithic config to a base `tsconfig.json` that I could extend, I wanted to verify changes via `showConfig`. Seeing no `files` there, no matter what I did, almost made me discard using that nice new `${configDir}` feature.

I can live with the current behaviour, now that I know. But maybe this will confuse other users as well, leading them down the wrong path too.

**Comment by stabpd** [comment]
> Will the absence of the `files` field have a substantial impact?

Nothing known to me, other than being confused.

> Or is it still there, just not displayed whe `tsc --showConfig` is executed?

Compilation works as expected even though the files are missing from `--showConfig`.
